### DISCUSSION TITLE
Grab metadata from an external DB instead of files

### DISCUSSION THREAD

**Discussion by shyonae** [navidrome/navidrome#2798] [Ideas]
Hi everyone!

I am using Navidrome (docker) and I love it, I just have one problem. I also use **beets** for metadata management, playlist enchantments and other things, but one thing beets does is modify the metadata **on the files directly**, I do this in order to have specific metadata on Navidrome, but to achieve this I have to copy the files from my download folder since I **do not want** to modify the source files.

Now, there is a _hardlink_ option, but that stores the metadata in an **external DB**, which Navidrome can't read (at least from what I checked these days)

I would even try and add the feature but I don't know where to start, I tried reading the code a bit but if was on my phone so I'll try to look at it a bit more. If you have any tips, I would appreciate them

Let me know what you think!

**Comment by shyonae** [comment]
Managed to get a simple rudimentary python script to work (do `pip install beets`):

This basically takes a .blb file from beets, which contains metadata for hardlinked files and spits them out, hope it can be useful for something:

```
import pickle
from beets.library import Library

library_path = './musiclibrary.blb'

library = Library(library_path)

# Get all items
items = library.items()

for item in items:
    print(item.keys())
    print(item.title)
    print(item.artist)
    print(item.album)
    print(item.genre)
    print(item.path)
    print()
```

**Comment by crims0nX** [comment]
Hey, I have interest on this feature too. I'm facing the same problem where I want to keep the original files pristine for archival purposes, but the tags aren't always in good shape, and for me to edit it I have to copy all the library.

If Navidrome gain the capability of loading the metadata from a external file, it will enable multiple workflows of long term data storage while also giving flexibility to tight-in the files to make Navidrome and Players happy.

`beets` has the [BPD plugin](https://beets.readthedocs.io/en/stable/plugins/bpd.html) that integrates a player into it, with it one can reproduce and navigate the beets' library through a web interface. A relevant excerpt from theirs documentation that I think it's worth mentioning: "_In the real MPD, the user can browse a music directory as it appears on disk. In beets, we like to abstract away from the directory structure. Therefore, BPD creates a “virtual” directory structure (artist/album/track) to present to clients._"

I tried to adopt beets' MPD, but it's lacking in functionality and integration that Navidrome has seen it's under _Subsonic_'s  ecosystem umbrella.

I think of it being something like: Create some sibling file alongside the original music one that would contain the metadata, or instead use a DB. Perhaps in beginning, write some documentation about how to edit the DB and let people do it "(semi-) manually" offering the user ways to edit it with a external tool (formats like JSON or YAML it's very common/popular and would be easy-sh to hand-roll) and, postpone the WebUI integration. With everything in place, it's use would be transparent by the Navidrome's clients.

Nice project btw :), I use it daily.

**Comment by cletusw** [comment]
https://github.com/navidrome/navidrome/discussions/3181 would likely solve this.
